docs/auth.md

```diff
@@ -0,0 +1,287 @@
+# Serving Certificates, Authentication, and Authorization
```

Comment:
I'm very happy to see the walkthrough has been updated to mention the Helm chart disables authn/authz by default, but could we also mention the same toward the top of this `auth.md` document so that anyone who's just kicking the tires won't waste too much time reading over this document. (Please don't mistake that a me undervaluing this document. This is _tremendously_ important for people who intend to use this in production and you have done a fantastic job of explaining things.)